Dev Final version for testing (#120)
digitalearthafrica/deafrica-airflow: dags/landsat-scenes-sync/README.md

# DEAfrica Landsat Scenes Sync

> Dags to read, either in bulk or daily, stacs released by USGS and migrate to Africa's servers

Reviewer: ```suggestion
Dags to identify USGS Collection 2 Landsat scenes either in bulk or per-day, push them onto a queue, and then to copy the data to Cape Town and upgrade the STAC metadata and create a notification.
```

Author: Done

## Development

### DAGs
- landsat-scenes-bulk

>   * This Dag will start a Python process which will download GZ files, which are split per landsat and collection, from USGS 

Reviewer: ```suggestion
  * This Dag will start a Python process which will download per-satellite tar.gz files from USGS'
```

Author: Done

>     file server, send the stac, in bulk, to our SQS queue in Africa. 

Reviewer: ```suggestion
    file server, send the STAC documents in bulk to our SQS queue in Africa. 
```

    
- landsat-scenes-daily

>   * This Dag will start a Python process which will access the USGS API, accordingly to informed dates, and 
>     send the stac to our SQS queue in Africa.

Reviewer: ```suggestion
  * This Dag will start a Python process which will access the USGS API, accordingly to informed dates, and 
    send the STAC documents to our SQS queue in Africa.
```

        
- landsat-scenes-process

>   * This Dag will be started by the scheduler to pull messages from the SQS queue, validate the message, process 
>     the message adding missing items, changing links to point to our S3 bucket and transforming from stac 0.7 to 1.0. 
>     After that sending to an SNS topic.

Reviewer: ```suggestion
  * This Dag will be started by the scheduler to pull messages from the SQS queue, validate the message, process 
    the message adding missing items, changing links to point to our S3 bucket and transforming from stac 0.7 to 1.0. 
    After that it sends the STAC document to an SNS topic.
```

Author: Done

  
- landsat-scenes-indexing

>   * This Dag will be started by the scheduler to pull the messages from a second SQS queue, which is 
>     filled automatically by the SNS, and will process by Datacube.  

Reviewer: ```suggestion
  * This Dag will be started by the scheduler to pull the messages from a second SQS queue, which is 
    filled automatically by the SNS, and will indexed into the Datacube.  
```

## Airflow Requirements

Install pip modules

```bash

>     pip install --user stactools[all] && pip install --user pystac && pip install --user    

Reviewer: ```suggestion
    pip install --user stactools[all] pystac
```

Author: Done
